src/dotnet-interactive-vscode/src/acquisition.ts

```diff
@@ -0,0 +1,129 @@
+    //   tool run dotnet-interactive --
+    // and both need to be tried
+    let localToolArguments = [
+        'tool',
+        'run',
+        'dotnet-interactive',
+        '--'
+    ];
+    let launchOptionSets = [
+        ['interactive'], // when launched as a global tool; working directory is irrelevant
+        localToolArguments // when launched as a local tool; working directory is possibly relevant
+    ];
+    let workingDirectory: string | undefined = undefined;
+    if (fs.existsSync(globalStoragePath)) {
+        // can't launch a process with a non-existent working directory
+        workingDirectory = globalStoragePath;
+    }
+    for (let launchOptions of launchOptionSets) {
+        let versionArguments = [...launchOptions];
+        versionArguments.push('--version');
+        let result = await processExitCodeAndOutput(dotnetPath, versionArguments, workingDirectory);
+        if (result.code === 0) {
+            // the interactive command was found, but maybe not the right version
+            const validDotnetInteractiveDevVersion: RegExp = /^.*-dev$/; // e.g., 1.0.0-dev
+            if (validDotnetInteractiveDevVersion.test(result.output)) {
+                // locally built versions are always assumed to be valid
+                return {
+                    args: launchOptions,
+                    workingDirectory
+                };
+            }
+
+            // otherwise we have to check the version
+            if (compareVersions.compare(result.output, versions.minimumDotNetInteractiveVersion, '>=')) {
+                return {
+                    args: launchOptions,
+                    workingDirectory
+                };
+            }
+        }
+    }
+
+    // if we got here there wasn't an appropriate version installed, so we'll install a local tool
+    // at this point the global storage path _must_ exist
+    // vs code doesn't guarantee that it exists, but does guarantee that we can create it
+    await vscode.workspace.fs.createDirectory(vscode.Uri.file(globalStoragePath));
+
+    // first ensure a tool manifest file exists
+    let localToolManifestFile = path.join(globalStoragePath, '.config', 'dotnet-tools.json');
+    if (!fs.existsSync(localToolManifestFile)) {
+        const { code } = await processExitCodeAndOutput(dotnetPath, ['new', 'tool-manifest', '--output', globalStoragePath]);
+        if (code !== 0) {
+            throw new Error('Unable to create local tool manifest file.');
+        }
+    }
+
+    // then acquire the tool and add it to the manifest
+    vscode.window.showInformationMessage('Acquiring latest `dotnet-interactive` tool.');
+    let toolSource = 'https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json';
+    const result = await processExitCodeAndOutput(dotnetPath, ['tool', 'install', '--add-source', toolSource, 'Microsoft.dotnet-interactive', '--tool-manifest', localToolManifestFile]);
+    if (result.code !== 0) {
```

Comment:
we should communicate both failure and success otherwise the users will never understand if we completed this or not